homework18/js/main.js

```diff
@@ -0,0 +1,196 @@
+const API_KEY = '91d38c9e';
+const searchInput = document.querySelector('.search-films__input');
+const searchButton = document.querySelector('.search-films__button');
+let pageNum;
+let films = [];
+
+//Функція помилки
+const notFoundError = (data) => {
+    const p = document.createElement('div');
+    p.innerText = `${data.Error}`;
+    p.classList.add('search-error');
+    document.querySelector('.list-films').append(p);
+}
+
+//Очишення блоків
+const deleteElements = (elementClassList) => {
+    return  document.querySelector(elementClassList).innerHTML = '';
+}
+
+//Показ фільмів
+const showMovie = (data) => {
+    if (data.Response === 'False') {
+        notFoundError(data);
+    } else {
+        data.Search.forEach((film) => {
+            const divFilmDescription = document.createElement('div');
+            divFilmDescription.classList.add('list-films__element');
+
+            //Додаємо зображення
+            const poster = document.createElement('img');
+            if (film.Poster === 'N/A') {
+                poster.src = '../assets/poster.jpg';
+            } else {
+                poster.src = `${film.Poster}`;
+            }
+            poster.alt = `Poster of "${film.Title}"`;
+
+            const divFilmInfo = document.createElement('div');
+            divFilmInfo.classList.add('list-films__info');
+
+            const titleOfFilm = document.createElement('p');
+            titleOfFilm.classList.add('list-films__description');
+            titleOfFilm.innerText = `${film.Title}, ${film.Year}, ${film.Type}`
+            divFilmInfo.append(titleOfFilm);
+
+            const btnDetails = document.createElement('button');
+            btnDetails.classList.add(`list-films__details`);
+            btnDetails.innerText = `Details`;
+            btnDetails.id = film.imdbID;
+
+            //Невеличку інформацію групуємо у окремий блок
+            divFilmDescription.append(poster, divFilmInfo, btnDetails);
+            document.querySelector('.list-films').append(divFilmDescription);
+        });
+    }
+}
+
+//Запит данних та виведення їх
+const getFilms = async () => {
+    deleteElements('.list-films');
+    deleteElements('.film-details');
+    const filmName = searchInput.value;
+    const URL = `http://www.omdbapi.com/?apikey=${API_KEY}&s=${filmName}&page=${pageNum}`;
+    try {
+        const filmList = await axios.get(URL);
+        films = filmList.data;
```

Comment:
Be careful with that copying. You need to deep-copy to make sure mutations won't affect